custom_components/unifiprotect/data.py

```diff
@@ -90,8 +91,22 @@ async def async_refresh(self, *_, force=False):
     def _async_process_ws_message(self, message: WSSubscriptionMessage):
         if message.new_obj.model in DEVICES_WITH_ENTITIES:
             self.async_signal_device_id_update(message.new_obj.id)
+        # trigger updates for camera that the event references
         elif isinstance(message.new_obj, Event) and message.new_obj.camera is not None:
             self.async_signal_device_id_update(message.new_obj.camera.id)
+        # trigger update for all viewports when a liveview updates
+        elif isinstance(message.new_obj, Liveview):
+            for viewer in self._protect.bootstrap.viewers.values():
+                self.async_signal_device_id_update(viewer.id)
+        # trigger update for all Cameras with LCD screens when NVR Doorbell settings updates
+        elif (
+            isinstance(message.new_obj, NVR)
+            and "doorbell_settings" in message.changed_data
+        ):
+            self._protect.bootstrap.nvr.update_all_messages()
+            for camera in self._protect.bootstrap.cameras.values():
+                if camera.feature_flags.has_lcd_screen:
+                    self.async_signal_device_id_update(camera.id)
```

Comment:
This is starting to get a bit long, can we avoid an `isinstance ` linear search?

Reply:
I cannot think of a good way. Each one is just different enough. The list is not likely to get any larger though.